testing/unittests/dalitests.cpp

```diff
@@ -3265,7 +3265,150 @@ class DaliSysInfoLoggerTester : public CppUnit::TestFixture
 CPPUNIT_TEST_SUITE_REGISTRATION( DaliSysInfoLoggerTester );
 CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( DaliSysInfoLoggerTester, "DaliSysInfoLoggerTester" );
 
+class FileReadPropertiesUpdaterTester : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE(FileReadPropertiesUpdaterTester);
+        CPPUNIT_TEST(testInit);
+        CPPUNIT_TEST(testFileReadPropertiesUpdater);
+    CPPUNIT_TEST_SUITE_END();
+
+    #define SCOPE "testfileupdate"
+    std::vector<std::string> subFileNames = { "subfile1", "subfile2", "subfile3" };
+    std::vector<std::string> superFileNames = { "superfile", "supersuperfile" };
+    std::string getSuperFileLogicalName(unsigned superfileNum)
+    {
+        return std::string(SCOPE) + "::" + superFileNames[superfileNum - 1];
+    }
+    std::string getSubFileLogicalName(unsigned subfileNum)
+    {
+        return std::string(SCOPE) + "::" + subFileNames[subfileNum - 1];
+    }
+    void cleanupTestFiles(bool assertOnFailure)
+    {
+        try
+        {
+            Owned<IDistributedSuperFile> supersuperfile = dir.lookupSuperFile(getSuperFileLogicalName(2).c_str(), user, AccessMode::writeMeta);
+            if (supersuperfile)
+            {
+                supersuperfile->removeSubFile(nullptr, false, false, nullptr);
```

Comment:
this isn't needed is it?
since this is only removing from a super that is about to be removed itself anyway.